SteamBot/Bot.cs

```diff
@@ -233,16 +254,12 @@ public bool StartBot()
             CreateLog();
             IsRunning = true;
 
-            log.Info("Connecting...");
-
-            if (!backgroundWorker.IsBusy)
-                // background worker is not running
-                backgroundWorker.RunWorkerAsync();
-
+            Log.Info("Connecting...");
+            if (botThread.IsBusy)
+                return true; //We will consider this running.
+            botThread.RunWorkerAsync();
```

Comment:
You have to change this as well.  I changed it back to

    if(!botThread.IsBusy)
        backgroundWorker.RunWorkerAsync();

With the `return true` as it is now, `SteamClient.Connect()` is never called..